File: templates/templates/javav2/tests/basic-http/AuthAdditionalTest.java
package org.openapis.openapi;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.Optional;

import org.junit.jupiter.api.Test;

import org.openapis.openapi.models.operations.BasicAuthOptionalSecurity;
import org.openapis.openapi.models.operations.BasicAuthOptionalResponse;
import org.openapis.openapi.models.components.Security;

public class AuthAdditionalTest {
    @Test
    void testBasicAuthOperationOptional() throws Exception {
        CommonHelpers.recordTest("auth-basic-auth-operation-optional");


        SDK s = SDK.builder()
                .security(Security.builder()
                        .username("wrongUser")
                        .password("wrongPass")
                        .build())
                .build();

        BasicAuthOptionalResponse res = s.auth().basicAuthOptional()
                .security(BasicAuthOptionalSecurity.builder()
                        .username("testUser")
                        .password("testPass")
                        .build())
                .call();
        assertNotNull(res);
        assertEquals(200, res.statusCode());
        assertTrue(res.basicAuthResponse().isPresent());
        assertTrue(res.basicAuthResponse().get().authenticated());
        assertEquals("testUser", res.basicAuthResponse().get().user());


        SDK s2 = SDK.builder()
                .security(Security.builder()
                        .username("testUser")
                        .password("testPass")
                        .build())
                .build();

        assertThrows(Exception.class, () -> {
            s2.auth().basicAuthOptional()
                    .security(BasicAuthOptionalSecurity.builder().build())
                    .call();
        });
    }
}

File: templates/templates/typescriptv2/tests/basic-http/auth_additional.test.ts
import { expect, test } from "vitest";

import { SDK } from "../index.js";
import { recordTest } from "./common_helpers.js";

test("Basic Auth Operation Optional", async () => {
  recordTest("auth-basic-auth-operation-optional");

  const sdk = new SDK({
    security: {
      username: "wrongUser",
      password: "wrongPass",
    },
  });

  const res = await sdk.auth.basicAuthOptional({
    username: "testUser",
    password: "testPass",
  });

  expect(res.httpMeta.response.status).toBe(200);
  expect(res.basicAuthResponse).toBeDefined();
  expect(res.basicAuthResponse!.authenticated).toBe(true);
  expect(res.basicAuthResponse!.user).toBe("testUser");

  const sdk2 = new SDK({
    security: {
      username: "testUser",
      password: "testPass",
    },
  });

  await expect(sdk2.auth.basicAuthOptional({})).rejects.toThrow();
});
